clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -581,7 +581,7 @@ public boolean hasUnflushedBatches() {
             if (!entry.getValue().isEmpty())
                 return true;
         }
-        return !this.incomplete.incomplete.isEmpty();
+        return !this.incomplete.isEmpty();
```

Comment:
We may want to move `IncompleteBatches` to its own file as a private package class so that we can't make this mistake again.